scripts/spack/packages/axom/package.py

```diff
@@ -241,7 +241,7 @@ def initconfig_compiler_entries(self):
         # Add optimization flag to workaround HIP compiler errors
         if "+rocm" in spec:
             if "crayCC" in self.compiler.cxx or spec.satisfies("%clang@16"):
-                entries.append(cmake_cache_string("CMAKE_CXX_FLAGS","-O1"))
+                entries.append(cmake_cache_string("CMAKE_CXX_FLAGS_DEBUG","-O1 -g -DNDEBUG"))
```

Comment:
Could you please comment on why we need to override the optimization flags for these compilers? Are these known bugs? Are the bugs in `CMake` or in the `clang16` and `crayCC` compilers?